### DISCUSSION TITLE
How to garantee the inspector order on the runtime inspector? it's not happening when have a object reference field

### DISCUSSION THREAD

**Discussion by riqmariz** [yasirkula/UnityRuntimeInspector#24] [Q&A] [answered]
I'm using this plugin, so game designers can inspect objects outside the project and generate jsons. But some objects that have a object reference field does not respect the order, it always goes first. How can I ensure to respect the order on the runtime inspector?

**Comment by riqmariz** [comment]
print of runtime inspector:
![image](https://user-images.githubusercontent.com/37335266/104048989-e99fc780-51c2-11eb-94d8-e49bb7ff13dc.png)

but it should be like this (with spawner as last field):
![image](https://user-images.githubusercontent.com/37335266/104049072-0e943a80-51c3-11eb-80d5-68219c2c3415.png)

**Reply by yasirkula** [reply]
Can you show me the _Spawn Group Name_ and _Spawner_ variables in the code? I'd like to see how and where they are defined in the code.

**Comment by riqmariz** [comment, marked as answer]
Thanks for the reply! 
I'm trying to put this as code on this comment, but I failed...

**here is the code of SpawnGroup:**
`
using UnityEngine;
using Random = UnityEngine.Random;

 [CreateAssetMenu(fileName="New SpawnGroup",menuName="Spawn Group")]
public class SpawnGroup : ScriptableObject
{
    [Header("Spawn Group")]
    [Tooltip("Name of the Group Selection")]
    [SerializeField]
    private string spawnGroupName;
    public string SpawnGroupName => spawnGroupName;

    [Header("Selection Enemy Of Group Settings")]
    [Tooltip("The group of ID enemies that will be used in the spawn")]
    [SerializeField]
    private GameObject[] enemySpawnGroup;
    public GameObject[] EnemySpawnGroup => enemySpawnGroup;
    
    [Tooltip("The selection in the enemy spawn group will be in order, from to first to the last")]
    [ConditionalHide( new[]{"randomEnemy","randomWeightedEnemy"},new[]{true,true},false,false)]
    [SerializeField]
    private bool inOrderEnemy = true;
    public bool InOrderEnemy => inOrderEnemy;
    
    [Tooltip("The selection in the enemy spawn group will be randomly")]
    [ConditionalHide( new[]{"inOrderEnemy","randomWeightedEnemy"},new[]{true,true},false,false)]
    [SerializeField]
    private bool randomEnemy = false;
    public bool RandomEnemy => randomEnemy;
    [Tooltip("The selection in the enemy spawn group will be randomly, but with weights associated")]
    [ConditionalHide( new[]{"inOrderEnemy","randomEnemy"},new[]{true,true},false,false)]
    [SerializeField]
    private bool randomWeightedEnemy = false;
    public bool RandomWeightedEnemy => randomWeightedEnemy;
    
    [Tooltip("The weights in the same order as the enemy spawn group, just used when random weighted enemy selected")]
    [SerializeField]
    private float[] weightsEnemy;

    public float[] WeightsEnemy => weightsEnemy;

    [Header("Selection Spawn Area Settings")]
    [Tooltip("The group of spawn area position that will be used in the spawn, the enum is associated to a screen position previous setted")]
    [SerializeField]
    private SpawnAreaPosition[] spawnPositions;
    public SpawnAreaPosition[] SpawnAreaPositions => spawnPositions;
    
    [Tooltip("The selection in the spawn area group will be in order, from to first to the last")]
    [ConditionalHide( new[]{"randomSpawnArea","randomWeightedSpawnArea"},new[]{true,true},false,false)]
    [SerializeField]
    private bool inOrderSpawn = true;

    public bool InOrderSpawn => inOrderSpawn;
    
    [Tooltip("The selection in the spawn area group will be randomly")]
    [ConditionalHide( new[]{"inOrderSpawn","randomWeightedSpawnArea"},new[]{true,true},false,false)]
    [SerializeField]
    private bool randomSpawnArea = false;

    public bool RandomSpawnArea => randomSpawnArea;
    
    [Tooltip("The selection in the spawn area group will be randomly, but with weights associated")]
    [ConditionalHide( new[]{"inOrderSpawn","randomSpawnArea"},new[]{true,true},false,false)]
    [SerializeField]
    private bool randomWeightedSpawnArea = false;

    public bool RandomWeightedSpawnArea => randomWeightedSpawnArea;
    
    [Tooltip("The weights in the same order as the spawn areas group, just used when random weighted enemy selected")]
    [SerializeField]
    private float[] weightsSpawnArea;

    public float[] WeightsSpawnArea => weightsSpawnArea;
    
    [Header("Choose your spawner")]
    [Tooltip("the spawn type that will be used to spawn the enemies")]
    [SerializeField]
    private Spawner spawner;

    public Spawner Spawner
    {
        get => spawner;
        set => spawner = value;
    }
 (...)
}
`

**Reply by yasirkula** [reply]
Related Issue: https://github.com/yasirkula/UnityRuntimeInspector/issues/19. You can see that it isn't possible to preserve the field-property-field order of the variables. In your case, however, you may wonder why the _spawner_ field, which is located at the bottom is drawn above all other fields. That's because RuntimeInspector draws the _Spawner_ property instead of the _spawner_ field here. RuntimeInspector draws properties (Spawner) above fields and hides properties' backing fields (spawner). Only Spawner property has both a getter and a setter and thus, it is the only property that is drawn above all other fields.

To workaround this feature, you can either change all your properties so that they have both a getter and a setter (it can be private), or remove Spawner property's setter.

**Reply by riqmariz** [reply]
Thanks!! I Understand now!
